From 367e1b3c77e787593fd52362163429c7112d1227 Mon Sep 17 00:00:00 2001
From: Nick Bofferding <bofferdn@us.ibm.com>
Date: Thu, 13 Oct 2016 15:36:56 -0500
Subject: [PATCH] Stop SBE scan service after building SMP

- Stop the SBE scan service after building SMP at end of istep 9.1
- Update istep dependencies to include HWP secureboot support
- Updated makefile to point to HWP secureboot support

Change-Id: Iae18a1838c7505e4f2eaecd03b7116dedf6d08c8
---
 src/include/usr/isteps/istep09list.H               |  3 ++
 .../hwpf/hwp/activate_powerbus/activate_powerbus.C | 60 ++++++++++++++++++++++
 src/usr/hwpf/hwp/activate_powerbus/makefile        |  3 +-
 3 files changed, 65 insertions(+), 1 deletion(-)

diff --git a/src/include/usr/isteps/istep09list.H b/src/include/usr/isteps/istep09list.H
index e68d004..3981773 100644
--- a/src/include/usr/isteps/istep09list.H
+++ b/src/include/usr/isteps/istep09list.H
@@ -79,6 +79,9 @@ namespace   INITSERVICE
 const DepModInfo  g_istep09Dependancies = {
     {
 #ifndef CONFIG_VPO_COMPILE
+        DEP_LIB(libslave_sbe.so),
+        DEP_LIB(libcore_activate.so),
+        DEP_LIB(libsecure_boot.so),
         DEP_LIB(libactivate_powerbus.so),
         DEP_LIB(libdram_initialization.so),
         DEP_LIB(libsbe.so),
diff --git a/src/usr/hwpf/hwp/activate_powerbus/activate_powerbus.C b/src/usr/hwpf/hwp/activate_powerbus/activate_powerbus.C
index 7879db3..0ec6e96 100644
--- a/src/usr/hwpf/hwp/activate_powerbus/activate_powerbus.C
+++ b/src/usr/hwpf/hwp/activate_powerbus/activate_powerbus.C
@@ -64,6 +64,9 @@
 #include    <pbusLinkSvc.H>
 
 #include    "proc_build_smp/proc_build_smp.H"
+#include    "proc_use_sbe_scan_service.H"
+#include    "proc_stop_sbe_scan_service.H"
+
 #include    <intr/interrupt.H>
 #include    <fsi/fsiif.H>
 
@@ -321,6 +324,63 @@ void*    call_proc_build_smp( void    *io_pArgs )
                 break;
             }
 
+            // The processor scan service can be forced on by
+            // attribute or enabled by security policy.  If the scan
+            // service is in use on the chip, disable it since it's no longer
+            // needed.  Not disabling can lead to unwanted PIB traffic that
+            // interferes with I2C master operations, etc.
+            fapi::Target fapiProcTarget(TARGET_TYPE_PROC_CHIP,
+                (const_cast<TARGETING::Target*>(l_proc_target)));
+
+            bool useScanService=false;
+            FAPI_INVOKE_HWP(l_errl, proc_use_sbe_scan_service,
+                            fapiProcTarget, useScanService);
+            if (l_errl)
+            {
+                TRACFCOMP(ISTEPS_TRACE::g_trac_isteps_trace,
+                    ERR_MRK " ERROR : proc_use_sbe_scan_service failed for "
+                    "PROC target 0x%08X",
+                    TARGETING::get_huid(l_proc_target));
+
+                // Create IStep error log and cross reference error that
+                // occurred
+                l_StepError.addErrorDetails(l_errl);
+
+                // Commit error
+                errlCommit( l_errl, HWPF_COMP_ID );
+                break;
+            }
+
+            if(useScanService)
+            {
+                // Note: we don't have access to the SBE image here, so if there
+                // is a failure we won't be able to capture all possible FFDC.
+                FAPI_INVOKE_HWP(l_errl, proc_stop_sbe_scan_service,
+                                fapiProcTarget,NULL);
+                if(l_errl)
+                {
+                    TRACFCOMP(ISTEPS_TRACE::g_trac_isteps_trace,
+                        ERR_MRK " ERROR : proc_stop_sbe_scan_service failed "
+                        "for PROC target 0x%08X",
+                        TARGETING::get_huid(l_proc_target));
+
+                    // Create IStep error log and cross reference error that
+                    // occurred
+                    l_StepError.addErrorDetails(l_errl);
+
+                    // Commit error
+                    errlCommit( l_errl, HWPF_COMP_ID );
+                    break;
+                }
+                else
+                {
+                    TRACFCOMP(ISTEPS_TRACE::g_trac_isteps_trace,
+                        INFO_MRK " SUCCESS : proc_stop_sbe_scan_service for "
+                        "PROC target 0x%08X",
+                        TARGETING::get_huid(l_proc_target));
+                }
+            }
+
             ++curproc;
         }
 
diff --git a/src/usr/hwpf/hwp/activate_powerbus/makefile b/src/usr/hwpf/hwp/activate_powerbus/makefile
index 0db637a..f6ee411 100644
--- a/src/usr/hwpf/hwp/activate_powerbus/makefile
+++ b/src/usr/hwpf/hwp/activate_powerbus/makefile
@@ -5,7 +5,7 @@
 #
 # OpenPOWER HostBoot Project
 #
-# Contributors Listed Below - COPYRIGHT 2012,2014
+# Contributors Listed Below - COPYRIGHT 2012,2016
 # [+] International Business Machines Corp.
 #
 #
@@ -39,6 +39,7 @@ EXTRAINCDIR += ${ROOTPATH}/src/usr/hwpf/hwp/bus_training
 EXTRAINCDIR += ${ROOTPATH}/src/usr/hwpf/hwp/edi_ei_initialization
 EXTRAINCDIR += ${ROOTPATH}/src/usr/hwpf/hwp/edi_ei_initialization/proc_fab_iovalid
 EXTRAINCDIR += ${ROOTPATH}/src/usr/hwpf/hwp/dram_initialization/proc_setup_bars
+EXTRAINCDIR += ${ROOTPATH}/src/usr/hwpf/hwp/secure_boot
 
 ##  NOTE: add the base istep dir here.
 EXTRAINCDIR += ${ROOTPATH}/src/usr/hwpf/hwp/activate_powerbus
-- 
1.8.2.2

